code/datums/perks/fate.dm

```diff
@@ -132,15 +132,32 @@
 		holder.stats.removePerk(src.type)
 		return
 	var/turf/T = get_turf(holder)
-	var/obj/item/W = pickweight(list(
+	var/obj/item/W = null
+	if(is_neotheology_disciple(holder) && prob(50))
+		W = pickweight(list(
+				/obj/item/tool/sword/nt/longsword = 0.5,
+				/obj/item/tool/sword/nt/shortsword = 0.5,
+				/obj/item/tool/sword/nt/scourge = 0.1,
+				/obj/item/tool/knife/dagger/nt = 0.5,
+				/obj/item/gun/energy/nt_svalinn = 0.4,
+				/obj/item/gun/energy/stunrevolver = 0.1))
+	else
+		W = pickweight(list(
+				/obj/item/tool/hammer/mace = 0.2,
+				/obj/item/tool/hammer/mace/makeshift/baseballbat = 0.1,
 				/obj/item/tool/knife/ritual = 0.5,
+				/obj/item/tool/knife/switchblade = 0.5,
 				/obj/item/tool/sword = 0.2,
 				/obj/item/tool/sword/katana = 0.2,
-				/obj/item/tool/knife/dagger/ceremonial = 0.8,
+				/obj/item/tool/knife/dagger = 0.5,
+				/obj/item/gun/projectile/colt = 0.2,
+				/obj/item/gun/projectile/revolver/havelock = 0.1,
+				/obj/item/tool/knife/dagger/ceremonial = 0.4,
 				/obj/item/gun/projectile/revolver = 0.4))
 	holder.sanity.valid_inspirations += W
 	W = new W(T)
 	W.desc += " It has been inscribed with the \"[holder.last_name]\" family name."
+	W.name = "[W] of [holder.last_name]"
```

Comment:
Does this interact well with weapon mods? I remember them using initial() but that might have been fixed.

Reply:
gun mods seem to be fine
tool mods however seem to cleanly override the new naming, which is sad but not the worst way that could go down.
Is there any decent fix to that?

Reply:
@hyperioo meant to ping you on above, sorry

Reply:
@MemedHams You could check how it's done for the gun to check why it's working properly with them and not with the tool. Toolmods are likely based on initial() value of `name` so they don't remember your custom name.